pkg/node/provisioner.go

```diff
@@ -1,40 +1,131 @@
 package node
 
 import (
+	"encoding/json"
 	"fmt"
 	"os"
 	"path/filepath"
-	"sync"
-)
+	"strings"
 
-var (
-	provisionerLock sync.Mutex
-	provisioner     = &provision{}
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/klog"
+	"k8s.io/utils/mount"
 )
 
-type provision struct {
-	Path string
+type Provisioner struct {
+	dataPath string
+	mounter  mount.Interface
+}
+
+func NewProvisioner(dataPath string) Provisioner {
+	return Provisioner{
+		dataPath: dataPath,
+		mounter:  mount.New(""),
+	}
+}
+
+func (p Provisioner) volPath(volID string) string {
+	return filepath.Join(p.dataPath, volID)
+}
+
+func (p Provisioner) createDir(volID string) error {
```

Comment:
have you considered a _fileutil_ pkg for the various helper funcs related to creating, removing directories and mount points?

Reply:
I wasn't too sure about adding more dependencies since I know core Kubernetes is conscious of the dependency chain. Is there a specific fileutil pkg that you would recommend?

Reply:
I meant your own fileutil package. A pkg to consolidate the various file related funcs.